Develop

krowly opened the pull request.

krowly committed:
Commit1

krowly committed:
Commit1

krowly committed:
Commit1

nikolay-977 commented on src/main/java/com/example/Lion.java:
```diff
@@ -4,19 +4,20 @@
 public class Lion {
 
+    Feline feline;
     boolean hasMane;
```
Можно улучшить: Для всех полей следует указывать минимально необходимые модификаторы доступа

nikolay-977 commented on src/test/java/AnimalParamTest.java:
```diff
@@ -0,0 +1,43 @@
+        if(animalKind.equals("Exception")) {
+            assertThrows(Exception.class, ()->animal.getFood(animalKind));
+        }
+        else
```
Необходимо исправить: Не следует использовать конструкцию if-else при написании тестов. Это затрудняет читаемость кода и может вызвать путаницу в понимании. Лучше реализовать 2 отдельных кейса

nikolay-977 commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,29 @@
+        public void LionTest1() throws Exception {
+            Lion Male = new Lion("Самец",feline);
+            Lion Female = new Lion("Самка",feline);
+            assertTrue(Male.doesHaveMane());
```
Необходимо исправить: не следует выполнять все проверки в одном кейсе. Иначе в случае фейла первой проверки, последующие проверки не будут выполнены, это может аффектить другие тесты, что увеличит время локализации дефекта. Следует написать отдельные кейсы для проверки каждого пола.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблице используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

krowly committed:
Внесены многочисленные изменения - начиная от добавления отдельного класса тестирования для исключений в классе Animal. Примечание - вероятно бы следовало вынести поля в классе тестов и тестов исключений, так как они совпадают.
Все assert и Mockito.verify методы были вынесены в отдельные кейсы.
Был заполнен файл Read.me по всему проекту, вероятно следовало бы сделать отдельный КeadME для тестов.
Исправлены нарушения конвенций об именах в некоторых местах.
В тестах хватает проблем из-за аннотаций, отсутствует единый стиль кода. В будущем постараюсь этого избегать. По хорошему надо было начинать везде с мока а затем писать spy или stub.
  Jacoco продолжает выдавать 100% покрытия, тесты исполняются без лишних проблем.
  В отчет был добавлен файл jacoc.exeс хотя не знаю что он делает.
  Также приношу свои извинения за долгие сроки, я болею физически.

  Благодарность ревьюверу за справедливую критику.

krowly commented:
Внесены многочисленные изменения - начиная от добавления отдельного к…
…ласса тестирования для исключений в классе Animal. Примечание - вероятно бы следовало вынести поля в классе тестов и тестов исключений, так как они совпадают.

Все assert и Mockito.verify методы были вынесены в отдельные кейсы.
Был заполнен файл Read.me по всему проекту, вероятно следовало бы сделать отдельный КeadME для тестов.
Исправлены нарушения конвенций об именах в некоторых местах.
В тестах хватает проблем из-за аннотаций, отсутствует единый стиль кода. В будущем постараюсь этого избегать. По хорошему надо было начинать везде с мока а затем писать spy или stub.
  Jacoco продолжает выдавать 100% покрытия, тесты исполняются без лишних проблем.
  В отчет был добавлен файл jacoc.exeс хотя не знаю что он делает.
  Также приношу свои извинения за долгие сроки, я болею физически.

  Благодарность ревьюверу за справедливую критику.
